proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+spec:
+  selector:
+    app: ml-pipeline-artifact-server
+  ports:
+  - port: 8080
+    protocol: TCP
+```
+
+##### Cross-Namespace Access Pattern
+
+Control plane components (UI, API server) in the `kubeflow` namespace need to access artifact servers in user namespaces:
+
+**Access Flow:**
+
+1. **UI determines target namespace**: Reads run metadata to identify which namespace contains the artifact
+2. **Service call**: UI calls `http://ml-pipeline-artifact-server.<namespace>.svc:8080/apis/v2beta1/...`
+3. **Authorization check**: Target namespace's artifact server performs `SubjectAccessReview`
+4. **Artifact retrieval**: If authorized, serves artifact from namespace's PVC
+
+Example: UI Accessing User Namespace Artifacts
+
+```go
+// UI backend code running in kubeflow namespace
+func (s *UIServer) GetArtifact(namespace, runID, nodeID, artifactName string) (*Artifact, error) {
+    // Construct cross-namespace URL
+    artifactURL := fmt.Sprintf(
+        "http://ml-pipeline-artifact-server.%s.svc:8080/apis/v2beta1/runs/%s/nodes/%s/artifacts/%s:read",
+        namespace, runID, nodeID, artifactName,
+    )
+    
+    // Request includes user's credentials for `SubjectAccessReview`
+    req, _ := http.NewRequest("GET", artifactURL, nil)
+    req.Header.Set("Authorization", userToken)
+    
+    // Artifact server in target namespace will:
+    // 1. Extract user identity from token
+    // 2. Perform `SubjectAccessReview` against pipelines.kubeflow.org/runs resource
+    // 3. Return artifact if user has access
+    resp, err := http.DefaultClient.Do(req)
+    // ...
+}
+```
+
+**Why Both Network Access + RBAC?**
+
+- **Network access** (`ml-pipeline-artifact-server.<namespace>.svc`): Enables cross-namespace communication via standard Kubernetes DNS
+- **RBAC authorization** (`SubjectAccessReview`): Ensures users can only access artifacts they have permissions for
+- **Defense in depth**: Network reachability doesn't imply authorization - both layers provide security
+
+This design allows the control plane to serve artifacts from any namespace while maintaining strict user-level access control.
+
+##### Direct Client Access
+
+In namespace-local mode, workflow pods connect directly to their namespace's artifact server:
+
+- Artifact URI: `kfp-artifacts://<namespace>/<pipeline>/<run-id>/...`
+- Resolved to: `http://ml-pipeline-artifact-server.<namespace>.svc.cluster.local:8080/...`
+- UI handles the resolution based on deployment mode
+
+##### Deployment Strategy
+
+1. **Proactive Provisioning**: The KFP API server watches for namespaces with `pipelines.kubeflow.org/enabled=true` annotation and creates:
+   - The artifact server Deployment
+   - The artifact server Service
+   - The namespace PVC (which only the artifact server will mount)
+2. **Lifecycle Management**: Artifact servers persist as long as the namespace exists
+3. **Resource Efficiency**: Minimal resource usage when idle (can scale to zero if configured)
```

Comment:
That is already in sync.py and should probably stay there. Why do you want to move it to the main api server ? Sync.py is for the pipelines profile controller to manage kfp enabled namespaces.